datasetinsights/evaluation_metrics/average_precision_2d.py

```diff
@@ -196,3 +185,105 @@ def n_point_interpolated_ap(recall, precision, point=11):
                 cc.append(p)
 
         return ap
+
+
+class AveragePrecisionIOU50(EvaluationMetric):
+    """2D Bounding Box Average Precision at IOU = 50%.
+
+    This implementation would calculate AP@50IOU for each label.
+
+    Attributes:
+        ap (AveragePrecision): AveragePrecision metrics
+    """
+
+    def __init__(self):
+        self.ap = AveragePrecision(iou_threshold=0.5)
+
+    def reset(self):
+        self.ap.reset()
+
+    def update(self, mini_batch):
+        self.ap.update(mini_batch)
+
+    def compute(self):
+        return self.ap.compute()
+
+
+class MeanAveragePrecisionIOU50(EvaluationMetric):
+    """2D Bounding Box Mean Average Precision metrics at IOU=50%.
+
+    This implementation would calculate mAP@50IOU. The metric can be
+    described as:
+    mAP = mean_{label}AP(label)@IOU50
```

Comment:
Have you checked documentation to see if this formula is rendered properly?

Reply:
Should add a space an `.. math::` flag for math formula to be parsed by latex properly.